packages/commonwealth/client/scripts/views/pages/Editor/Editor.tsx

```diff
@@ -0,0 +1,262 @@
+    },
+    [handleDropAsync],
+  );
+
+  const handleDragEnter = useCallback((event: React.DragEvent) => {
+    console.log('handleDragStart');
+    event.preventDefault();
+    dragCounterRef.current = dragCounterRef.current + 1;
+
+    if (dragCounterRef.current === 1) {
+      setDragging(true);
+    }
+  }, []);
+
+  const handleDragOver = useCallback((event: React.DragEvent) => {
+    console.log('handleDragOver');
+
+    event.preventDefault();
+    // This is necessary to allow a drop
+    event.dataTransfer!.dropEffect = 'copy'; // Shows a copy cursor when dragging files
+  }, []);
+
+  const handleDragLeave = useCallback((event: React.DragEvent) => {
+    console.log('handleDragLeave');
+    event.preventDefault();
+    dragCounterRef.current = dragCounterRef.current - 1;
+
+    if (dragCounterRef.current === 0) {
+      setDragging(false);
+    }
+  }, []);
+
+  return (
+    <div
+      className={clsx(
+        'mdxeditor-container',
+        'mdxeditor-container-mode-' + mode,
+      )}
+      onDrop={handleDrop}
+      onDragEnter={handleDragEnter}
+      onDragOver={handleDragOver}
+      onDragLeave={handleDragLeave}
+    >
+      <MDXEditor
+        onError={errorHandler}
+        ref={mdxEditorRef}
+        markdown={supported}
+        placeholder={placeholder}
+        iconComponentFor={iconComponentFor}
+        translation={(key, defaultValue, interpolations) => {
+          switch (key) {
+            case 'toolbar.blockTypeSelect.placeholder':
+              // show the default placeholder that's active here..
+              return 'H1';
+            case 'toolbar.blockTypes.heading':
+              if (interpolations?.level) {
+                // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+                return 'H' + interpolations.level;
+              }
+              return 'H1';
+            case 'toolbar.blockTypes.quote':
+              return 'Q';
+            case 'toolbar.blockTypes.paragraph':
+              return 'P';
+            default:
+              return defaultValue;
+          }
+        }}
+        plugins={[
+          toolbarPlugin({
+            location: mode === 'mobile' ? 'bottom' : 'top',
+            toolbarContents: () =>
+              mode === 'mobile' ? <ToolbarForMobile /> : <ToolbarForDesktop />,
+          }),
+          listsPlugin(),
+          quotePlugin(),
+          headingsPlugin(),
+          linkPlugin(),
+          linkDialogPlugin(),
+          codeBlockPlugin({ defaultCodeBlockLanguage: 'js' }),
+          codeMirrorPlugin({
+            codeBlockLanguages,
+          }),
+          imagePlugin({ imageUploadHandler }),
+          tablePlugin(),
+          thematicBreakPlugin(),
+          frontmatterPlugin(),
+          diffSourcePlugin({ viewMode: 'rich-text', diffMarkdown: 'boo' }),
+          markdownShortcutPlugin(),
+        ]}
+      />
+
+      {mode === 'desktop' && (
+        <DesktopEditorFooter onImportMarkdown={handleImportMarkdown} />
```

Comment:
If the editor has some text and user imports a markdown file, what is the expected behavior here
1. Replace text in editor with imported markdown
2. Keep text in editor and add the imported markdown text.

Right now, its 1, but I think we should go for 2 or if we want to keep 1 add a warning message that editor text will be removed

cc: @burtonator @masvelio